api/src/main/java/org/pmiops/workbench/institution/InstitutionServiceImpl.java

```diff
@@ -438,10 +438,23 @@ private void setInstitutionTierRequirement(
       throw new ServerErrorException(
           "Failed to cleanup existing tier requirements before replacing them");
     }
-    institutionTierConfigMapper
-        .tierConfigsToDbTierRequirements(
-            modelInstitution.getTierConfigs(), dbInstitution, dbAccessTiers)
-        .forEach(institutionTierRequirementDao::save);
+
+    // If requirement is NO_ACCESS, it is possible that this tier does not exist yet. Skip the
```

Comment:
This does not cover the case where:
1. the tier exists
2. the inst previously provided access
3. the inst switches to NO_ACCESS

(please add a test case for this)

Instead, we can filter on the tier's existence directly (pseudocode: `filter(dbAccessTiers.contains(tier))`) before we decide to save the ITR to the DB.

Reply:
The logic now becomes if NO_ACCESS from UI, API will just delete the old one(the 1st line in this method). 
Then insert noting (the code I just added). Added a test for that.
Not sure if the logic for filter(dbAccessTiers.contains(tier))) is correct. e.g. if CT not exist, but user choose to set CT to DOMAINS, shall we just ignore it or error out?

Reply:
sorry, I missed the pre-deletion step.

Since we have that ... yes, not saving it when NO_ACCESS is a good solution.